File: data_processor.py
# Unused import
import json

# Hardcoded credentials (security hotspot)
API_KEY = "sk_live_1234567890abcdefghijklmnop"

# Unused variable
CONFIG_SETTINGS = {"timeout": 30}

# Function with potential division by zero
def calculate_statistics(numbers):
    total = sum(numbers)
    return total / len(numbers)  # Possible ZeroDivisionError

# Duplicated code (code smell)
def process_data_v1(data):
    return [x * 2 for x in data]

def process_data_v2(data):
    return [x * 2 for x in data]  # Duplicate of v1

# Inconsistent naming
def fetch_user_info(userId):  # camelCase
    return {"id": userId, "name": "Test User"}

# Unreachable code
def deprecated_method():
    print("This method is no longer used")
    return True
deprecated_method()
print("This message is unreachable")

# SQL injection vulnerability
def get_user_query(user_id):
    query = f"SELECT * FROM users WHERE id = {user_id}"  # Security hotspot
    return query

# Poor error handling
def parse_value(value):
    return int(value)  # Potential ValueError

# Unused function
def helper_method():
    pass

if __name__ == "__main__":
    # Demonstrate issues
    print(calculate_statistics([]))  # Empty list
    print(parse_value("text"))  # Invalid conversion
    print(get_user_query("1 OR 1=1"))  # SQL injection demo

File: DataProcessor.java
package com.example;

import java.util.List;

// Unused import
import java.util.ArrayList;

public class DataProcessor {
    // Hardcoded credentials (security hotspot)
    private static final String API_KEY = "prod_1234567890abcdef";

    // Unused field
    private int unusedField = 42;

    // Duplicated code (code smell)
    public List<Integer> processDataV1(List<Integer> data) {
        return data.stream().map(x -> x * 2).toList();
    }

    public List<Integer> processDataV2(List<Integer> data) {
        return data.stream().map(x -> x * 2).toList(); // Duplicate of V1
    }

    // Potential division by zero
    public double calculateAverage(List<Integer> numbers) {
        int sum = numbers.stream().mapToInt(Integer::intValue).sum();
        return sum / numbers.size();
    }

    // Inconsistent naming
    public String fetchUserInfo(String userId) { // camelCase parameter
        return "User: " + userId;
    }

    // Unused method
    private void helperMethod() {
        System.out.println("This is never called");
    }

    // Null handling issue
    public int getStringLength(String input) {
        return input.length(); // Potential NPE
    }

    // SQL injection vulnerability
    public String createUserQuery(int userId) {
        return "SELECT * FROM users WHERE id = " + userId; // Security hotspot
    }

    public static void main(String[] args) {
        DataProcessor processor = new DataProcessor();
        System.out.println(processor.calculateAverage(List.of())); // Empty list
        System.out.println(processor.getStringLength(null)); // Null input
    }
}
